service/src/test/java/bio/terra/workspace/db/DbRetryUtilsTest.java

```diff
@@ -0,0 +1,85 @@
+package bio.terra.workspace.db;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import bio.terra.workspace.common.BaseUnitTest;
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.dao.CannotSerializeTransactionException;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+
+public class DbRetryUtilsTest extends BaseUnitTest {
+
+  static String FAKE_QUERY = "INSERT INTO (fake_row, other_fake_row) VALUES (:foo, :bar)";
+  static MapSqlParameterSource FAKE_PARAMS =
+      new MapSqlParameterSource().addValue("foo", "foo").addValue("bar", "bar");
+  static RowMapper<Integer> FAKE_ROW_MAPPER = (rs, rowNum) -> 42;
+
+  private static final CannotSerializeTransactionException RETRY_EXCEPTION =
+      new CannotSerializeTransactionException("test");
+
+  @Mock NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Test
+  void updateSucceedsWithRetry() {
+    when(jdbcTemplate.update(any(), (SqlParameterSource) any()))
+        .thenThrow(RETRY_EXCEPTION)
+        .thenReturn(42);
```

Comment:
Oh neat, I didn't realize you could have multiple then statements like this.

Reply:
it's pretty handy - I just learned about `thenThrow(array)` this PR too, which is pretty cool.